lib/keychain.js

```diff
@@ -0,0 +1,236 @@
+const { generateDeterministicKeyPair, deriveAddress, hash, curveToID, hashAlgoToID, IDToCurve, IDToHashAlgo }= require("./crypto")
+const { uint8ArrayToHex, concatUint8Arrays} = require("./utils")
+const { createHmac, createHash } = require("crypto")
+
+module.exports.newKeychain = newKeychain
+module.exports.decodeKeychain = decodeKeychain
+module.exports.keyToJWK = keyToJWK
+
+class Keychain {
```

Comment:
Keychain may integrate a transaction builder, or transaction builder may be updated so we can build transactions and sign them with keychain services.
Actually ```TxBuilder.build()``` can only take a seed and an index so we cannot sign transaction with our new Keychain using services
We may create a new issue to add this feature as it is consequent to implement ?

Reply:
Good idea, but there is a function for this purpose `setPreviousSignatureAndPreviousPublicKey` brought by @prix-uniris, using in the ledger-cli app. Because hardware, clients could leverage the tx builder, but use signature in a secure environment. 

So it would just impact the way to build a transaction, by using three steps:
1. Build the transaction with builder 
2. Leverage keychain to generate the keys
3. Add the key materials to the builder.